File: libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/ConfirmUsernameRequest.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ConfirmUsernameRequest {
  @JsonProperty
  private String usernameHash;

  @JsonProperty
  private String zkProof;

  @JsonProperty
  private String encryptedUsername;

  public ConfirmUsernameRequest(String usernameHash, String zkProof, String encryptedUsername) {
    this.usernameHash      = usernameHash;
    this.zkProof           = zkProof;
    this.encryptedUsername = encryptedUsername;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/ConfirmUsernameRequest.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

class ConfirmUsernameRequest(
  val usernameHash: String,
  val zkProof: String,
  val encryptedUsername: String
)
